### ISSUE TITLE
Babe applies exponential lenience, complaing there was no block for two slots

### ISSUE THREAD

**Issue by eskimor** [paritytech/substrate#10738]
although the last block got validated 6 seconds ago.

This happens when a block gets reverted as a result of a dispute, so it might just be that babe considers the reverted to block as the last seen block, which causes this.

![babe-weirdness](https://user-images.githubusercontent.com/1527017/151198254-a49440e7-df7c-443d-999d-815c5b8c4404.png)

[Logs](https://grafana.parity-mgmt.parity.io/explore?orgId=1&left=%5B%221643185901000%22,%221643185909000%22,%22loki.parity-testnet%22,%7B%22refId%22:%22A%22,%22expr%22:%22%7Bpod%3D%5C%22versi%2Fversi-validator-a-node-6%5C%22%7D%20%7C~%5C%22chain-selection%7Cbabe%5C%22%22%7D%5D)

@andresilva what do you think?

**Comment by eskimor** [comment]
A little while later it is even worse (claims to not have seen a block for 3 slots):
![babe-3slots](https://user-images.githubusercontent.com/1527017/151204488-012c07dc-b97a-4bab-8a42-50c1ac9dbf3e.png)

**Comment by bkchr** [comment]
Slots are measured in time and not blocks. 

Meaning if you revert 3 blocks back, there will be 3 slots "missed". We can not really do anything against this, but I also don't see any real problem with this. We have initially added this mechanic for "big blocks", like blocks that executed a large runtime upgrade and didn't make it in time. The block would always be produced, because we do not really abort any call to the runtime, but we would not send it out to the network because the authorship already has moved on. With the slot lenience factor kicking in, we will just give the authorship more time to wait for the block production to finish. Now, back to normal operation mode (assuming no big runtime upgrade) and an increased slot time, we will still end the block in the allowed 2 seconds, because we still have weights in place that ensure that we can not push more transactions than the block weight allow.

TLDR: Yes, the slot time is increased after a revert, but nothing problematic should happen.

**Comment by eskimor** [comment]
Thanks Basti! That makes a lot of sense and explains a few things.
